payments_service/base/tests/__init__.py

```diff
@@ -107,6 +107,7 @@ def endpoint(self, view):
         dynamic_urls.urlpatterns = patterns(
             '',
             url(r'^dynamic-endpoint$', view),
+            url(r'^dynamic-endpoint/(?P<pk>[^/]+)?/?$', view),
```

Comment:
It would, could probably just tighten up the URL views, its just for testing. Basically if you use the second URL every Django view being tested needs to accept pk in it, because the regex group is applied to the kwargs of the method.